asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py

```diff
@@ -0,0 +1,311 @@
+            )
+        else:
+            logger.warning(
+                "There are usernames corresponding to multiple user_ids that are not Deleted Users."
+            )
+
+    check_id_username2 = mse_data.groupby("user_id")[["username"]].nunique()
+
+    ids_corresponding_to_multiple_usernames = check_id_username2[
+        check_id_username2["username"] > 1
+    ]
+    if len(ids_corresponding_to_multiple_usernames) > 0:
+        logger.warning(
+            f"There are {len(ids_corresponding_to_multiple_usernames)} user IDs corresponding to multiple usernames."
+        )
+
+
+def plot_number_interactions_per_user(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a histogram with number of interactions per user.
+    The computation is based on all MSE posts, not necessarily the ones we collected data for.
+
+    Args:
+        mse_data (pd.DataFrame): MSE category/sub-forum
+    """
+
+    interactions_per_user_overall = mse_data[["user_id", "n_posts_user"]]
+
+    # due to data collection at different times, numbers might differ, so we need to remove duplicates for each user
+    interactions_per_user_overall.sort_values("n_posts_user", ascending=False)
+    interactions_per_user_overall.drop_duplicates("user_id", keep="first", inplace=True)
+
+    plt.figure(figsize=(pc.figsize_x, pc.figsize_y))
+    plt.hist(
+        interactions_per_user_overall["n_posts_user"],
+        bins=30,
+        color=pc.NESTA_COLOURS[0],
+        edgecolor="white",
+    )
+    plt.xlabel("Number of posts and replies")
+    plt.ylabel("Number of users")
+    plt.title("Number of users with a given\nnumber of posts and replies")
+    plt.yscale("log")
+    plt.tight_layout()
+    path_to_plot = f"{MSE_FIGURES_PATH}/category_{category}_interactions_per_user.png"
+    plt.savefig(path_to_plot)
+    logger.info(
+        f"Plot with number of interactions per user for category '{category}' saved successfully in {path_to_plot}"
+    )
+
+
+def plot_number_posts_per_user(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a histogram with number of posts per user.
+    The computation is based on data collected (i.e. posts collected from the specified category,
+    not the whole MSE)
+
+    Args:
+        mse_data (pd.DataFrame): MSE category/sub-forum
+    """
+    posts_per_user_data_collected = (
+        mse_data[mse_data["is_original_post"] == 1].groupby("user_id")[["id"]].nunique()
+    )
+
+    posts_per_user_data_collected.reset_index(inplace=True)
+    posts_per_user_data_collected.columns = ["user", "n_posts_user"]
+
+    plt.figure(figsize=(pc.figsize_x, pc.figsize_y))
+    plt.hist(
+        posts_per_user_data_collected["n_posts_user"],
+        bins=range(0, 90, 5),
+        color=pc.NESTA_COLOURS[0],
+        edgecolor="white",
+    )
+    plt.xlabel("Number of posts")
+    plt.ylabel("Number of users")
+    plt.title("Number of users with a given number of posts\n(in data collected)")
```

Comment:
```suggestion
    plt.title(f"Number of users with a given number of posts\nin {category} category")
```
It might be a clearer title

Reply:
I do need to add the category in there - thanks for the reminder! And maybe that's all we really need to make this less confusing. Because basically:
- when you collect the data, user information comes w/ total number of posts/replies by a user (not just in the forum where collecting, but across all MSE);
- Based on the data I collected, I can compute the number of users w/ a certain number of posts (what I'm doing here)